--- postgresql-9.3.1/src/backend/executor/nodeModifyTable.c	2013-10-08 03:17:38.000000000 +0000
+++ postgresql-9.3.1-mod/src/backend/executor/nodeModifyTable.c	2013-11-13 06:01:55.689057384 +0000
@@ -674,8 +674,10 @@
 		result = heap_update(resultRelationDesc, tupleid, tuple,
 							 estate->es_output_cid,
 							 estate->es_crosscheck_snapshot,
-							 true /* wait for commit */ ,
+							 false /* wait for commit */ ,
 							 &hufd, &lockmode);
+
+
 		switch (result)
 		{
 			case HeapTupleSelfUpdated:
@@ -743,7 +745,7 @@
 				return NULL;
 
 			default:
-				elog(ERROR, "unrecognized heap_update status: %u", result);
+			  //elog(ERROR, "unrecognized heap_update status: %u", result);
 				return NULL;
 		}
 
